galasa-parent/dev.galasa.framework.api.runs/src/main/java/dev/galasa/framework/api/runs/internal/PropertyStore.java

```diff
@@ -0,0 +1,69 @@
+package dev.galasa.framework.api.runs.internal;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+import javax.servlet.Servlet;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ServiceScope;
+
+import dev.galasa.framework.spi.IConfigurationPropertyStoreService;
+import dev.galasa.framework.spi.IFramework;
+import dev.galasa.framework.spi.utils.GalasaGsonBuilder;
+
+@Component(service = Servlet.class, scope = ServiceScope.PROTOTYPE, property = {
+"osgi.http.whiteboard.servlet.pattern=/propertystore" }, name = "Galasa CPS")
+public class PropertyStore extends HttpServlet {
+    private static final long serialVersionUID = 1L;
+
+    private Log logger = LogFactory.getLog(getClass());
+
+    private final Gson gson = GalasaGsonBuilder.build();
+
+    @Reference
+    public IFramework framework; // NOSONAR
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+
+        JsonObject requestBody = gson.fromJson(new InputStreamReader(req.getInputStream()), JsonObject.class);
```

Comment:
Would rather these parameters be requested via URL parameters rather than JSON bodies.   That way you can make the request via a browser.